docker_publisher.py

```diff
@@ -0,0 +1,299 @@
+                                       namespaces=REPOMD_NAMESPACES)
+        primary_req = requests.get(self.repourl + "/" + path_primary)
+        return zlib.decompress(primary_req.content, zlib.MAX_WBITS | 32)
+
+    def getRPMUrl(self, pkgname, arch):
+        primary_tree = xml.fromstring(self.fetchPrimaryXml())
+        pkgs = primary_tree.xpath("md:package[./md:name/text() = '%s']" % (pkgname),
+                                  namespaces=REPOMD_NAMESPACES)
+
+        for pkg in pkgs:
+            if arch in pkg.xpath("./md:arch/text()", namespaces=REPOMD_NAMESPACES):
+                return self.repourl + "/" + pkg.xpath("./md:location/@href",
+                                                      namespaces=REPOMD_NAMESPACES)[0]
+
+    def getDockerImage(self, callback):
+        """Download and extract the RPM from the OSS repo."""
+        image_layer_file = tempfile.NamedTemporaryFile(delete=False)
+
+        rpm_url = self.getRPMUrl("opensuse-tumbleweed-image", self.arch)
+        if rpm_url is None:
+            raise DockerFetchException("Could not get the URL for the RPM package")
+
+        with tempfile.NamedTemporaryFile() as rpm_file:
+            rpm_file.write(requests.get(rpm_url).content)
+            with tempfile.TemporaryDirectory() as tar_dir:
+                # Extract the .tar.xz inside the RPM into the dir
+                subprocess.call("rpm2cpio '%s' | cpio -i --quiet --to-stdout \*.tar.xz | tar -xJf - -C '%s'" % (rpm_file.name, tar_dir), shell=True)
+                # Parse the manifest to get the name of the root tar.xz
+                manifest = json.loads(open(tar_dir + '/manifest.json').read())
+                layers = manifest[0]['Layers']
+                if len(layers) != 1:
+                    raise DockerFetchException("Unexpected count of layers in the image")
+
+                image_layer_file = tar_dir + '/' + layers[0]
+                return callback(image_layer_file)
+
+
+def run():
+    tw_fetcher = DockerImageFetcherTW(versioned_redir="http://download.opensuse.org/tumbleweed/iso/openSUSE-Tumbleweed-DVD-x86_64-Current.iso",
+                                      repourl="http://download.opensuse.org/tumbleweed/repo/oss/suse",
+                                      arch="x86_64")
+    tw_publisher = DockerImagePublisherGit(git_path=os.path.expanduser("~/docker-containers-build"),
+                                           git_branch="openSUSE-Tumbleweed",
+                                           path="x86_64", arch="x86_64")
+
+    try:
+        current = tw_fetcher.currentVersion()
+        released = tw_publisher.releasedDockerImageVersion()
+    except Exception as e:
+        print("Exception during version fetching - aborting:")
+        raise e
+
+    print("Available version: %s" % (current))
+    print("Released version: %s" % (released))
+
+    release = current != released
+
+    if "--release" in sys.argv[1:]:
```

Comment:
Take a look at `argparse` (used in this repo for quick examples) and easy to use.

Reply:
Will do, once there actually is a proper CLI interface.

Reply:
Why wait, add two lines and you have a real arg parsing.